PAWS/ECGRecordings/ECGModule.swift

```diff
@@ -24,27 +25,60 @@ import UserNotifications
 
 @Observable
 class ECGModule: Module, DefaultInitializable, EnvironmentAccessible {
-    // periphery:ignore - The ConfigureFirebaseApp injection is required to enforce an initialization within Spezi before this module.
-    @ObservationIgnored @Dependency(ConfigureFirebaseApp.self) private var firebaseConfiguration
+    @ObservationIgnored @Dependency(Account.self) private var account: Account?
+    @ObservationIgnored @Dependency(AccountNotifications.self) private var accountNotifications: AccountNotifications?
     @ObservationIgnored @Dependency(HealthKit.self) private var healthKit
+    @ObservationIgnored @AppStorage(StorageKeys.healthKitStartDate) var healthKitStartDate: Date?
+    
     
     private(set) var electrocardiograms: [HKElectrocardiogram] = []
-    private var authStateDidChangeListenerHandle: AuthStateDidChangeListenerHandle?
     private let healthStore = HKHealthStore()
     private let logger = Logger(subsystem: "PAWS", category: "ECGModule")
+    private var notificationsTask: Task<Void, Never>?
+    
+    
+    private var healthKitSamplesEndDateCutoff: Date {
+        get async throws {
+            // Waiting until Spezi Account loads the account details.
+            let loadingStartDate = Date.now
+            while await account?.details?.dateOfEnrollment == nil || loadingStartDate.distance(to: .now) > 2.0 {
+                logger.debug("Loading DateOfEnrollment ...")
+                try await Task.sleep(for: .seconds(0.05))
+            }
```

Comment:
@Supereg As noted in the meeting: I have to unfortunately add this workaround here to access the date of enrollment in the Standard here; I am wondering if there is a way to guarantee this on init of the account type instead of a delayed call as we have the sync loading of account details in the recent versions of SpeziAccount & Firebase.